### PR TITLE
Update Flow graph in GUI for VictoriaMetrics #5295

### PR DESCRIPTION
**Pull request by IvanChupin**
BFF: new API to acquire the stats for victoria metrics + rest client to get the victoria metrics from VictoriaDb + prepare victoria data.
UI: add switch checkbox to the flow-details page to chose the source  of stats whether it be victoria or opentsdb. + added corresponding calls to acquire stats data victoria/tsdb.

This PR also includes minor UI and BFF code style improvements.

The Metrics enum already exists. I just moved it to a separate file. It probably doesn't have the best design, but redesigning it would be time-consuming since I would need to change all the related logic."

**env parameters:**
```
upd: kilda_victoriametrics_port -> kilda_victoriametrics_write_port;
add: kilda_victoriametrics_read_port
```
#5295

### CODE REVIEW

File: src-gui/src/test/java/org/openkilda/service/StatsServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@ExtendWith(MockitoExtension.class)
+class StatsServiceTest {
```
Comment (pablomuri) [inline src-gui/src/test/java/org/openkilda/service/StatsServiceTest.java:52]: I would add more tests trying bad values to check the error handling.

**Review by pablomuri — CHANGES_REQUESTED** [review]
Just two comments. Good job!

File: src-gui/src/main/java/org/openkilda/service/StatsService.java [hunk 2]
```diff
@@ -63,149 +73,143 @@ public class StatsService {
...
+     * @throws InvalidRequestException if there is an issue with the request parameters.
+     */
+    public List<VictoriaData> getTransformedFlowVictoriaStats(String statsType,
+                                                              String startDate,
+                                                              String endDate,
+                                                              String step,
+                                                              String flowId,
+                                                              List<String> metric,
+                                                              Direction direction) throws InvalidRequestException {
+        return getFlowVictoriaStats(statsType, startDate, endDate, step, flowId, metric, direction);
```
[the remark this answers is not in the document]
Comment (IvanChupin) [reply src-gui/src/main/java/org/openkilda/service/StatsService.java]: Agree, thanks, fixed

File: src-gui/src/test/java/org/openkilda/service/StatsServiceTest.java (hunk 1 above)
Comment (IvanChupin) [reply src-gui/src/test/java/org/openkilda/service/StatsServiceTest.java:53]: Thanks, added more corner cases for tests.

**Review by pablomuri — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Comment by pkazlenka** [comment]
@IvanChupin , I can't build the branch locally. I receive the next errors:
```
ERROR in app/modules/flows/flow-datatables/flow-datatables.component.ts:270:21 - error TS2300: Duplicate identifier 'copyItem'.

270   copyToClip(event, copyItem, index) {
                        ~~~~~~~~
app/modules/flows/flow-datatables/flow-datatables.component.ts:271:11 - error TS2300: Duplicate identifier 'copyItem'.

271     const copyItem = this.clipBoardItems[index][copyItem];
              ~~~~~~~~
app/modules/flows/flow-list/flow-list.component.ts:137:27 - error TS2448: Block-scoped variable 'statusParam' used before its declaration.

137       const statusParam = statusParam.filter(function (el) {
                              ~~~~~~~~~~~

  app/modules/flows/flow-list/flow-list.component.ts:137:13
    137       const statusParam = statusParam.filter(function (el) {
                    ~~~~~~~~~~~
    'statusParam' is declared here.
app/common/services/flowpath.service.ts:269:14 - error TS2304: Cannot find name 'scale'.

269       .scale(scale)
                 ~~~~~
app/common/services/flowpath.service.ts:270:17 - error TS2304: Cannot find name 'translate'.

270      .translate(translate[0], translate[1]);
                    ~~~~~~~~~
app/common/services/flowpath.service.ts:270:31 - error TS2304: Cannot find name 'translate'.

270      .translate(translate[0], translate[1]);
                                  ~~~~~~~~~
app/modules/flows/flow-path/flow-path.component.ts:239:40 - error TS2304: Cannot find name 'src_switch'.

239     this.flowpathService.loadIslDetail(src_switch, src_port, dst_switch, dst_port);
                                           ~~~~~~~~~~
app/modules/flows/flow-path/flow-path.component.ts:239:52 - error TS2304: Cannot find name 'src_port'.

239     this.flowpathService.loadIslDetail(src_switch, src_port, dst_switch, dst_port);
                                                       ~~~~~~~~
app/modules/flows/flow-path/flow-path.component.ts:239:62 - error TS2304: Cannot find name 'dst_switch'.

239     this.flowpathService.loadIslDetail(src_switch, src_port, dst_switch, dst_port);
                                                                 ~~~~~~~~~~
app/modules/flows/flow-path/flow-path.component.ts:239:74 - error TS2304: Cannot find name 'dst_port'.

239     this.flowpathService.loadIslDetail(src_switch, src_port, dst_switch, dst_port);
                                                                             ~~~~~~~~
app/modules/flows/flow-path/flow-path.component.ts:251:44 - error TS2552: Cannot find name 'svgElement'. Did you mean 'SVGElement'?

251     this.flowpathService.zoomFn(direction, svgElement, type);
                                               ~~~~~~~~~~

  ../node_modules/typescript/lib/lib.dom.d.ts:12815:13
    12815 declare var SVGElement: {
                      ~~~~~~~~~~
    'SVGElement' is declared here.
app/modules/flows/flow-path/flow-path.component.ts:271:13 - error TS2304: Cannot find name 'flowSwitches'.

271         if (flowSwitches.indexOf(dataCommon[d].switch_id) != -1) {
                ~~~~~~~~~~~~
app/modules/switches/switch-datatable/switch-datatable.component.ts:338:43 - error TS2304: Cannot find name 'copyData'.

338     this.clipboardService.copyFromContent(copyData);
                                              ~~~~~~~~
app/modules/reports/affected-flow-list/affected-flow-list.component.ts:195:21 - error TS2300: Duplicate identifier 'copyItem'.

195   copyToClip(event, copyItem, index) {
                        ~~~~~~~~
app/modules/reports/affected-flow-list/affected-flow-list.component.ts:196:11 - error TS2300: Duplicate identifier 'copyItem'.

196     const copyItem = this.clipBoardItems[index][copyItem];
              ~~~~~~~~
app/modules/reports/affected-flows-report/affected-flows-report.component.ts:72:27 - error TS2448: Block-scoped variable 'statusParam' used before its declaration.

72       const statusParam = statusParam.filter(function (el) {
                             ~~~~~~~~~~~

  app/modules/reports/affected-flows-report/affected-flows-report.component.ts:72:13
    72       const statusParam = statusParam.filter(function (el) {
                   ~~~~~~~~~~~
    'statusParam' is declared here.
app/modules/topology/world-map-view/world-map-view.component.ts:667:11 - error TS2304: Cannot find name 'style'.

667             return style;
                       ~~~~~

make[1]: *** [Makefile:11: .deps/node] Error 1
make[1]: Leaving directory '/home/ubuntu/open-kilda/src-gui'
make: *** [generated.mk:101: compile-common] Error 2
```
Do you have any idea how to fix the problem?

**Review by dmitrii-beliakov — APPROVED** [review]
